src/ray/raylet/worker.cc

```diff
@@ -138,23 +137,15 @@ void Worker::AssignTask(const Task &task, const ResourceIdSet &resource_id_set,
       task.GetTaskExecutionSpec().GetMessage());
   request.set_resource_ids(resource_id_set.Serialize());
 
-  auto status = rpc_client_->AssignTask(request, [](Status status,
-                                                    const rpc::AssignTaskReply &reply) {
+  return rpc_client_->AssignTask(request, [](Status status,
```

Comment:
Should this just be a RAY_CHECK_OK? Since it's an async call.

Reply:
I don't think so. That would cause the raylet to crash if a worker dies unexpectedly, right? In the caller here we disconnect the worker and retry submission if the call fails.

Reply:
I meant that async calls should always return status::OK.